src/GitHub.Exports.Reactive/Collections/TrackingCollection.cs

```diff
@@ -0,0 +1,842 @@
+            return -1;
+        }
+
+        /// <summary>
+        /// The unfiltered has a lazy cache that gets filled
+        /// up when something is looked up.
+        /// </summary>
+        /// <param name="item"></param>
+        /// <returns></returns>
+        int GetIndexUnfiltered(T item)
+        {
+            int ret = -1;
+            if (!sortedIndexCache.TryGetValue(item, out ret))
+            {
+                ret = original.IndexOf(item);
+                if (ret >= 0)
+                    sortedIndexCache.Add(item, ret);
+            }
+            return ret;
+        }
+
+        /// <summary>
+        /// When items get moved/inserted/deleted, update the indexes in the cache.
+        /// If start < end, we're inserting an item and want to shift all the indexes
+        /// between start and end to the right (+1)
+        /// If start > end, we're removing an item and want to shift all
+        /// indexes to the left (-1).
+        /// </summary>
+        static void UpdateIndexCache(int start, int end, IList<T> list, Dictionary<T, int> indexCache)
+        {
+            var change = end < start ? -1 : 1;
+            for (int i = start; i != end; i += change)
+                if (indexCache.ContainsKey(list[i]))
+                    indexCache[list[i]] += change;
+        }
+
+        static int FindNewPositionForItem(int idx, bool lower, IList<T> list, Func<T, T, int> comparer, Dictionary<T, int> indexCache)
+        {
+            var i = idx;
+            if (lower) // replacing element has lower sorting order, find the correct spot towards the beginning
+                for (var pos = i - 1; i > 0 && comparer(list[i], list[pos]) < 0; i--, pos--)
+                {
+                    Swap(list, i, pos);
+                    SwapIndex(list, i, 1, indexCache);
+                }
+
+            else // replacing element has higher sorting order, find the correct spot towards the end
+                for (var pos = i + 1; i < list.Count - 1 && comparer(list[i], list[pos]) > 0; i++, pos++)
+                {
+                    Swap(list, i, pos);
+                    SwapIndex(list, i, -1, indexCache);
+                }
+            indexCache[list[i]] = i;
+            return i;
+        }
+
+        /// <summary>
+        /// Swap two elements
+        /// </summary>
+        static void Swap(IList<T> list, int left, int right)
+        {
+            var l = list[left];
+            list[left] = list[right];
+            list[right] = l;
+        }
+
+        static void SwapIndex(IList<T> list, int pos, int change, Dictionary<T, int> cache)
+        {
+            if (cache.ContainsKey(list[pos]))
+                cache[list[pos]] += change;
+        }
+
+        static int BinarySearch(List<T> list, T item, Func<T, T, int> comparer)
+        {
+            return list.BinarySearch(item, new LambdaComparer<T>(comparer));
+        }
+
+        static IScheduler GetScheduler(IScheduler scheduler)
```

Comment:
Actually, it is used if `DISABLE_REACTIVEUI` is set. We should probably wrap this method in the same conditional.